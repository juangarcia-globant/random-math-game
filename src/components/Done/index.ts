export { default, Done } from './Done';