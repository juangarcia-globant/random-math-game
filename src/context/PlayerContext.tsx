import { useState, createContext, useContext, ReactNode } from 'react';

type PlayerProviderProps = {children: ReactNode};

const PlayerStateContext = createContext<
{username: string, setUsername: React.Dispatch<React.SetStateAction<string>>} | undefined
>(undefined);

function PlayerProvider ({ children }: PlayerProviderProps) {
  const [username, setUsername] = useState<string>('');

  return (
    <PlayerStateContext.Provider value={{ username, setUsername }}>
      { children }
    </PlayerStateContext.Provider>
  );
}

function usePlayerContext () {
  const context = useContext(PlayerStateContext);

  if (context === undefined) {
    throw new Error('usePlayerContext must be inside a PlayerProvider');
  }

  return context;
}

export { PlayerProvider, usePlayerContext };