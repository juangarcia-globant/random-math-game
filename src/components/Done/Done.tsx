// Components
import Button from "../Button";
// Context
import { usePlayerContext } from '../../context/PlayerContext';

import './Done.css';

type DoneProps = {
  message: string,
  onClick: () => void,
};

export const Done: React.FunctionComponent<DoneProps> = ({
  message = '',
  onClick,
}) => {
  const { username } = usePlayerContext();

  return (
    <div>
      <h2 className='username'> { username } </h2>
      <h1> {message} </h1>
      <Button text="Play Again" onClick={onClick} />
    </div>
  )
}

export default Done;