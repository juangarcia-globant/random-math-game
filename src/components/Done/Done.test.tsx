import { render, screen } from "@testing-library/react";
import { PlayerProvider } from "../../context/PlayerContext";
import Done from "./Done";

describe("<Done />", () => {
  test("Matches the snapshot", () => {
    const { asFragment } = render(
      <PlayerProvider>
        <Done message="Game over" onClick={() => {}} />
      </PlayerProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Should render GAME OVER text", () => {
    render(
      <PlayerProvider>
        <Done message="Game over" onClick={() => {}} />
      </PlayerProvider>
    );

    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toBe(` Game over `)
    expect(screen.getByRole('button', { name: "Play Again" })).toBeInTheDocument();
  });
});
