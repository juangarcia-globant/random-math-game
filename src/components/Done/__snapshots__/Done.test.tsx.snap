// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Done /> Matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="username"
    >
        
    </h2>
    <h1>
       Game over 
    </h1>
    <button
      class="button"
    >
      Play Again
    </button>
  </div>
</DocumentFragment>
`;
